i implemented changerequests from @merks and copilot.

the resulting message looks like this now:
```
Jun 09, 2026 9:23:55 AM org.eclipse.birt.report.engine.util.ResourceLocatorWrapper findResource
WARNING: http://localhost:3080/logo/test/logo.svg is not accessible.
java.io.IOException: {"success":false,"message":"logo not found"}
	at org.eclipse.birt.report.engine.util.ResourceLocatorWrapper.findResource(ResourceLocatorWrapper.java:97)
	at org.eclipse.birt.report.engine.emitter.ImageReader.readImage(ImageReader.java:260)
	at org.eclipse.birt.report.engine.emitter.ImageReader.readImage(ImageReader.java:250)
	at org.eclipse.birt.report.engine.emitter.ImageReader.read(ImageReader.java:129)
	at org.eclipse.birt.report.engine.nLayout.area.impl.ImageAreaLayout.initialize(ImageAreaLayout.java:93)
	at org.eclipse.birt.report.engine.nLayout.area.impl.ImageAreaLayout.layout(ImageAreaLayout.java:84)
	at org.eclipse.birt.report.engine.nLayout.LayoutEngine.startContent(LayoutEngine.java:426)
	at org.eclipse.birt.report.engine.emitter.ContentEmitterAdapter.startImage(ContentEmitterAdapter.java:181)
	at org.eclipse.birt.report.engine.emitter.ContentEmitterUtil.startContent(ContentEmitterUtil.java:86)
	at org.eclipse.birt.report.engine.layout.html.buffer.HTMLPageBuffer.startContent(HTMLPageBuffer.java:102)
	at org.eclipse.birt.report.engine.layout.html.buffer.TableBreakBuffer.startContent(TableBreakBuffer.java:226)
	at org.eclipse.birt.report.engine.layout.html.HTMLLeafItemLM.start(HTMLLeafItemLM.java:65)
	at org.eclipse.birt.report.engine.layout.html.HTMLAbstractLM.layout(HTMLAbstractLM.java:132)
	at org.eclipse.birt.report.engine.layout.html.HTMLBlockStackingLM.layoutNodes(HTMLBlockStackingLM.java:66)
	at org.eclipse.birt.report.engine.layout.html.HTMLPageLM.layout(HTMLPageLM.java:92)
	at org.eclipse.birt.report.engine.layout.html.HTMLReportLayoutEngine.layout(HTMLReportLayoutEngine.java:97)
	at org.eclipse.birt.report.engine.presentation.ReportDocumentBuilder.build(ReportDocumentBuilder.java:226)
	at org.eclipse.birt.report.engine.api.impl.RunTask.doRun(RunTask.java:224)
	at org.eclipse.birt.report.engine.api.impl.RunTask.run(RunTask.java:108)
	at org.eclipse.birt.report.designer.ui.preview.editors.AbstractViewer.createReportDocument(AbstractViewer.java:90)
	at org.eclipse.birt.report.designer.ui.viewer.StaticHTMLViewer.renderReport(StaticHTMLViewer.java:666)
	at org.eclipse.birt.report.designer.ui.viewer.StaticHTMLViewer$16.work(StaticHTMLViewer.java:761)
	at org.eclipse.birt.report.designer.ui.viewer.job.AbstractJob.run(AbstractJob.java:48)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.io.IOException: Server returned HTTP response code: 500 for URL: http://localhost:3080/logo/bimme/logo.svg
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:2028)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1629)
	at org.eclipse.birt.report.engine.util.ResourceLocatorWrapper.findResource(ResourceLocatorWrapper.java:84)
	... 23 more
```

--- a/engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/util/ResourceLocatorWrapper.java
+++ b/engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/util/ResourceLocatorWrapper.java
@@ -17,10 +17,14 @@ package org.eclipse.birt.report.engine.util;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.HttpURLConnection;
 import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.logging.Level;
+import java.util.logging.LogRecord;
 import java.util.logging.Logger;
 
 import org.eclipse.birt.report.engine.i18n.MessageConstants;
@@ -74,15 +78,32 @@ public class ResourceLocatorWrapper {
 			}
 			byte[] inBytes = cache.get(url);
 			if (inBytes == null) {
+				URLConnection connection = null;
 				try {
-					InputStream in = url.openStream();
-					inBytes = getByteArrayFromInputStream(in);
-					in.close();
-					cache.put(url, inBytes);
+					connection = url.openConnection();
+					try (InputStream in = connection.getInputStream()) {
+						inBytes = getByteArrayFromInputStream(in);
+						cache.put(url, inBytes);
+					}
 				} catch (IOException e) {
-					logger.log(Level.WARNING, MessageConstants.RESOURCE_NOT_ACCESSIBLE, url.toExternalForm());
+					LogRecord record = new LogRecord(Level.WARNING, MessageConstants.RESOURCE_NOT_ACCESSIBLE);
+					record.setParameters(new Object[] { url.toExternalForm() });
+					record.setLoggerName(logger.getName());
+					record.setResourceBundle(logger.getResourceBundle());
+					record.setResourceBundleName(logger.getResourceBundleName());
+					if (connection instanceof HttpURLConnection httpConn) {
+						String errorBody = extractErrorBody(httpConn);
+						if (!errorBody.isEmpty()) {
+							record.setThrown(new IOException(errorBody, e));
+						}
+					}
+					logger.log(record);
 					cache.put(url, DUMMY_BYTES);
 					return DUMMY_BYTES;
+				} finally {
+					if (connection instanceof HttpURLConnection httpConn) {
+						httpConn.disconnect();
+					}
 				}
 			}
 			return inBytes;
@@ -100,6 +121,25 @@ public class ResourceLocatorWrapper {
 		buffer = out.toByteArray();
 		out.close();
 		return buffer;
+	}
+
+	private String extractErrorBody(HttpURLConnection httpConn) {
+		try {
+			InputStream errorStream = httpConn.getErrorStream();
+			if (errorStream == null) {
+				return "";
+			}
+
+			try (errorStream) {
+				String body = new String(errorStream.readAllBytes(), StandardCharsets.UTF_8).trim();
+				if (body.isEmpty()) {
+					return "";
+				}
+				return body.length() > 1024 ? body.substring(0, 1024) + "..." : body;
+			}
+		} catch (Exception ex) {
+			return "";
+		}
 	}
 
 }

